Tmedia 660 prettier eslint

JackHowa opened the pull request:
## Description

Use prettier 

## Jira Ticket
- [TMEDIA-660](https://arcpublishing.atlassian.net/browse/TMEDIA-660)

## Acceptance Criteria
- Prettier fix all necessary files
- Setup git step for formatting changes

## Test Steps

1. Checkout this branch `git checkout TMEDIA-660-prettier-eslint`
2. Run `npm run lint`. Run prettier `npm run format`. don't see any conflicts
3. Tests pass as before. (I updated the snapshots as there were styled-component differences.)

## Effect Of Changes
### Before
Inconsistent code style

### After
More consistent code style

## Dependencies or Side Effects
- Saw some irregular whitespace linting change on one file where I disabled it with eslint. This seemeed to be the result of using inline snapshot
- Updated eslint as there were some requirements for updating that from other prettier packages

## Author Checklist
_The author of the PR should fill out the following sections to ensure this PR is ready for review._
- [x] Confirmed all the test steps a reviewer will follow above are working.
- [x] Confirmed there are no linter errors. Please run `npm run lint` to check for errors. Often, `npm run lint:fix` will fix those errors and warnings.
- [ ] Ran this code locally and checked that there are not any unintended side effects. For example, that a CSS selector is scoped only to a particular block.
- [ ] Confirmed this PR has reasonable code coverage. You can run `npm run test:coverage` to see your progress. -> na
  - [ ] Confirmed this PR has unit test files
  - [ ] Ran `npm run test`, made sure all tests are passing
  - [ ] If the amount of work to write unit tests for this change are excessive,
please explain why (so that we can fix it whenever it gets refactored).
- [x] Confirmed relevant documentation has been updated/added. -> update vs code in editor suggestions and documentation

## Reviewer Checklist
_The reviewer of the PR should copy-paste this template into the review comments on review._

- [ ] Linting code actions have passed.
- [ ] Ran the code locally based on the test instructions.
  - [ ] I don’t think this is needed to be tested locally. For example, a padding style change (storybook?) or a logic change (write a test).
- [ ] I am a member of the engine theme team so that I can approve and merge this. If you're not on the team, you won't have access to approve and merge this pr.
- [ ] Looked to see that the new or changed code has code coverage, specifically. We want the global code coverage to keep on going up with targeted testing.

JackHowa committed:
ci: Add basic eslint and prettier integrtion steps

JackHowa committed:
style: Fix eslint issue with space

/Users/howardj/sites/fusion-theme-blocks/.eslintrc.js   39:5  error  Expected exception block, space or tab after '//' in comment  spaced-comment

JackHowa committed:
fix: Don't use prettier on locale files

matthewroach commented on blocks/top-table-list-block/features/top-table-list/default.scss:
```diff
@@ -1,129 +1,128 @@
+			padding-top: map-get($spacers, "xs");
+			@media screen and (min-width: map-get($grid-breakpoints, "md")) {
+				padding-top: 0;
+				margin-top: -map-get($spacers, "xs");
```
This is causing an regression shown in chromatic, we will need to wrap the `map-get()` call in `()` 
```suggestion
				margin-top: -(map-get($spacers, "xs"));
```

It's a known issue with prettier - https://github.com/prettier/prettier/issues/4698

JackHowa replied:
wrong link? nvm found the conversation

matthewroach replied:
Updated - sorry about that

JackHowa replied:
no worries, had found it. updated. no ui regressions (thank you storybook/chromatic)

matthewroach commented on blocks/top-table-list-block/features/top-table-list/default.scss:
```diff
@@ -1,129 +1,128 @@
+		.md-promo-headline {
+			@media screen and (min-width: map-get($grid-breakpoints, "md")) {
+				margin-bottom: map-get($spacers, "xs");
+				margin-top: -map-get($spacers, "xs");
```
Same issue as noted above with prettier issue

JackHowa replied:
good point, will investigate

JackHowa committed:
ci: Setup husky precommit format hook

JackHowa committed:
style: Use prettier print preferred width of 100

Based on discussion in other pr https://github.com/WPMedia/arc-themes-components/pull/24

JackHowa committed:
ci: Use vscode built in code action settings with eslint and prettier rec'd

JackHowa committed:
style: Use prettier to enforce tabs, new lines, and other consistent formatting

JackHowa commented:
One change seems to only be a false positive
![Screen Shot 2022-02-23 at 10 56 29](https://user-images.githubusercontent.com/5950956/155368000-7c5dc08b-d574-4ab8-9837-8a9307ee9eb5.png)

JackHowa commented:
@visionbegin @nschubach can you review this as well?

nschubach commented:
Is this a decision we made?
![image](https://user-images.githubusercontent.com/2287238/155559720-86f7b8f7-1618-422b-8a36-704346416b51.png)

nschubach commented on .storybook/manager-head.html:
```diff
@@ -1,44 +1,44 @@
 <!-- reference for customizing https://github.com/storybookjs/storybook/issues/11222#issuecomment-646411961 -->
-<link rel="stylesheet" href="https://nav-prod.home.static.arcpublishing.com/css/arcnav.css">
+<link rel="stylesheet" href="https://nav-prod.home.static.arcpublishing.com/css/arcnav.css" />
```
`<link>` being a [void element](https://www.w3.org/TR/2011/WD-html-markup-20110113/syntax.html#void-elements) (or an [Empty Element](https://developer.mozilla.org/en-US/docs/Glossary/Empty_element)) doesn't really need the closing slash since it can never have children.  `<link></link>` is invalid, therefore `<link />` is also technically invalid, but the spec ignores self closing tags on void elements.  So it's a toss up whether we keep this.

JSX requires closing, so things like `img` would need it in React.  Closing marks were required under XHTML 1.0, but HTML5 does not.  I don't think we need to revert this, but I thought I'd share.

nschubach reviewed (approved):
Just a few comments, otherwise I think these are fine and it works as expected.
